packages/realm/src/Realm.ts

```diff
@@ -511,16 +511,24 @@ export class Realm {
       );
       const table = binding.Helpers.getTable(this.internal, objectSchema.tableKey);
       table.removeObject(obj.key);
+    } else if (subject instanceof List) {
+      const internalList = subject.internal;
+      internalList.removeAll();
+    } else if (subject instanceof Results) {
+      const internalResult = subject.internal;
+      internalResult.clear();
```

Comment:
Little need for a local variable.
```suggestion
      subject.internal.clear();
```